s3fs/core.py

```diff
@@ -454,6 +454,32 @@ def copy(self, path1, path2):
             raise IOError('Copy failed', (path1, path2))
         self.invalidate_cache(buc2)
 
+    def bulk_delete(self, pathlist):
+        """
+        Remove multiple keys with one call
+
+        Parameters
+        ----------
+        pathlist : listof strings
+            The keys to remove, must all be in the same bucket.
+        """
+        buckets = {split_path(path)[0] for path in pathlist}
+        if len(buckets) > 1:
+            raise ValueError("Bulk delete files should refer to only one bucket")
+        bucket = buckets.pop()
+        if len(pathlist) > 1000:
+            for i in range((len(pathlist) // 1000) + 1):
+                print(i)
+                self.bulk_delete(pathlist[i*1000:(i+1)*1000])
+            return
```

Comment:
Yeah, not suggesting a change.  Just pointing out a function I find I use fairly often.